homeassistant/components/climate/velbus.py

```diff
@@ -0,0 +1,71 @@
+"""
+Support for Velbus thermostat.
+
+For more details about this platform, please refer to the documentation
+https://home-assistant.io/components/climate.velbus/
+"""
+import logging
+
+from homeassistant.components.climate import (
+    STATE_HEAT, SUPPORT_TARGET_TEMPERATURE, ClimateDevice)
+from homeassistant.components.velbus import (
+    DOMAIN as VELBUS_DOMAIN, VelbusEntity)
+from homeassistant.const import ATTR_TEMPERATURE
+
+_LOGGER = logging.getLogger(__name__)
+
+DEPENDENCIES = ['velbus']
+
+SUPPORT_FLAGS = (SUPPORT_TARGET_TEMPERATURE)
+
+
+async def async_setup_platform(
+        hass, config, async_add_entities, discovery_info=None):
+    """Set up the Velbus thermostat platform."""
+    if discovery_info is None:
+        return
+
+    sensors = []
+    for sensor in discovery_info:
+        module = hass.data[VELBUS_DOMAIN].get_module(sensor[0])
+        channel = sensor[1]
+        sensors.append(VelbusClimate(module, channel))
+
+    async_add_entities(sensors)
+
+
+class VelbusClimate(VelbusEntity, ClimateDevice):
+    """Representation of a Velbus thermostat."""
+
+    @property
+    def supported_features(self):
+        """Return the list off supported features."""
+        return SUPPORT_FLAGS
+
+    @property
+    def temperature_unit(self):
+        """Return the unit this state is expressed in."""
+        return self._module.get_unit(self._channel)
```

Comment:
This should return either `TEMP_CELSIUS` or `TEMP_FAHRENHEIT` based on the device response. It's best to decouple the two APIs.

Reply:
the problem is that the system can have both values, depending on the config

Reply:
How do you mean?

Reply:
we can configure the devices to send the value as celcius or as farenheit, so we don't know this up front. so we have to ask the module for it

Reply:
Yes, this is fine. I'd just like us to map the response to either of the two home assistant constants. Ok?

Reply:
Done